src/Common/src/Interop/Unix/System.Native/Interop.Poll.cs

```diff
@@ -50,14 +48,17 @@ internal struct pollfd
         /// Returns a positive number (which is the number of structures with nonzero revent files), 0 for a timeout or no 
         /// descriptors were ready, or -1 on error.
         /// </returns>
-        internal unsafe static int poll(int fd, PollFlags flags, int timeout, out PollFlags resultFlags)
+        internal static int Poll(int fd, PollFlags flags, int timeout, out PollFlags resultFlags)
         {
-            pollfd pfd = default(pollfd);
-            pfd.fd = fd;
-            pfd.events = flags;
-            int result = poll(&pfd, 1, timeout);
-            resultFlags = pfd.revents;
-            return result;
+            unsafe
```

Comment:
If we're going to move the ```unsafe``` out of the signature, we should probably make it as tight as possible around what actually needs to be unsafe, which is just the call to Poll, right?